### PR TITLE
chore: Add RTL tests for modal and overlay

### PR DESCRIPTION
**Pull request by jakemhiller**
Broke out these tests from #806 because of the addition of react-testing-library

### PR Checklist

- [ ] Related to Abstract designs:
- [ ] Related to JIRA ticket: ABC-123
- [ ] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### Description

---

## Merging your changes

When you are ready to merge to master and publish your changes, use the "squash and merge" button in GitHub, and follow the [PR Title Guide](https://github.com/Codecademy/client-modules#pr-title-guide) to format your PR title and write your PR merge description.

**IMPORTANT:** If your PR contains breaking changes, please remember to follow the instructions for breaking changes!

### CODE REVIEW

File: script/jest/rtl-setup.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+import '@testing-library/jest-dom';
```
Comment (jakemhiller) [inline script/jest/rtl-setup.ts:1]: I don't think we've used this before, but it seems like one of the only ways to write tests with meaningful output from RTL. Thoughts? https://github.com/testing-library/jest-dom

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.7.4-alpha.65102c.0<br/>@codecademy/gamut@7.9.1-alpha.65102c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5ec360da8cb58497368711e0--codecademy-styleguide.netlify.app>https://5ec360da8cb58497368711e0--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5ec360da8cb58497368711e0>Deploy Logs</a></p>

File: script/jest/rtl-setup.ts (hunk 1 above)
Comment (jakemhiller) [reply script/jest/rtl-setup.ts:1]: Also, by making this a `.ts` file, we also get TS types on `expect` for the new methods.

Comment (codecaaron) [reply script/jest/rtl-setup.ts:1]: Makes sense in the context of portal dependent components.   I like it.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.7.4-alpha.0ff6da.0<br/>@codecademy/gamut@7.9.1-alpha.0ff6da.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5ec467377b0b534ed5281a31--codecademy-styleguide.netlify.app>https://5ec467377b0b534ed5281a31--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5ec467377b0b534ed5281a31>Deploy Logs</a></p>

**Pull request merged**